pennylane/ops/qubit/matrix_ops.py

```diff
@@ -86,7 +86,14 @@ def decomposition(U, wires):
             decomp_ops = qml.transforms.decompositions.zyz_decomposition(U, wire)
             return decomp_ops
 
-        raise NotImplementedError("Decompositions only supported for single-qubit unitaries")
+        if qml.math.shape(U) == (4, 4):
+            wires = Wires(wires)
+            decomp_ops = qml.transforms.decompositions.two_qubit_decomposition(U, wires)
```

Comment:
@josh146 so doing this here actually messes up the behaviour when `tape.expand()` is called, because of the order in which the operations are created (the centre part gets computed before the SU(2) stuff around). However if I call this with `invisible` around it, nothing happens during `expand` because nothing gets queue. It seems kind of weird to loop through the `decomp_ops` and queue them here, is there a different workaround for this?

Reply:
Huh, this is a really good question 🤔

Reply:
```python
      if qml.math.shape(U) == (4, 4):
          wires = Wires(wires)

          decomp_ops = qml.transforms.invisible(
              qml.transforms.two_qubit_decomposition
          )(U, wires)

          current_tape = qml.tape.get_active_tape()

          if current_tape:
              for op in decomp_ops:
                  qml.apply(op, current_tape)

          return decomp_ops
```

Huh, this feels gross, but it works.

Reply:
My mind is going blank, all I can think about is
> loop through the decomp_ops and queue them here

Also, it means that users won't be able to use `qml.transforms.decompositions.two_qubit_decomposition(U, wires)` directly either, since the order will be wrong.

I suppose what you could do is call the utility functions *invisibly*, *inside* `two_qubit_decomposition`, in order to generate a list of operators in the correct order. And then, right before returning, `apply()` each op in order? This is exactly the same as what you suggest above, just 'inside' `two_qubit_decomposition`.

Reply:
> Huh, this feels gross, but it works.

😆

Reply:
^ What you suggest is the nicest. It further simplifies `unitary_to_rot` so that the 2-qubit case looks just like the 1-qubit case.